ast/format.go

```diff
@@ -376,9 +376,31 @@ func (f *formatter) formatIdentifier(n *Identifier) {
 }
 
 func (f *formatter) formatStringLiteral(n *StringLiteral) {
-	f.writeRune('"')
-	f.writeString(n.Value)
-	f.writeRune('"')
+	if n.Loc != nil && n.Loc.Source != "" {
+		// Preserve the exact literal if we have it
+		f.writeString(n.Loc.Source)
+	} else {
+		f.writeRune('"')
+		f.writeString(escapeStr(n.Value))
+		f.writeRune('"')
+	}
+}
+
+func escapeStr(s string) string {
+	if !strings.ContainsAny(s, `"\"`) {
```

Comment:
Is this the correct argument to this? This appears to have `"` in the string twice. I think this should be `"\`.